test/integration/tcp_conn_pool_integration_test.cc

```diff
@@ -187,4 +221,76 @@ TEST_P(TcpConnPoolIntegrationTest, MultipleRequests) {
   tcp_client->close();
 }
 
+TEST_P(TcpConnPoolIntegrationTest, PoolCleanupEnabled) {
```

Comment:
another one I'd suggest is a test of shutting down the server with active connections, to make sure the drain logic plays well with shutdown logic.